### DISCUSSION TITLE
Remove entries from the progress bar later in training.

### DISCUSSION THREAD

**Discussion by ChristophReich1996** [Lightning-AI/pytorch-lightning#13390] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi everyone,

I have the following "issue" I have multiple training phases with different loss functions. However, I would only like to show the current loss values in my progress bar. Is there any elegant way to discard old loss entries (eg. loss_bootstrap_1) from the progress bar?
![image](https://user-images.githubusercontent.com/34400551/175323421-7fe8e8cd-0a68-429d-b21f-89508e06c8c4.png)

Thanks for the help! :)

Cheers
Christoph

**Comment by akihironitta** [comment]
Hi @ChristophReich1996! Not sure if it's the best (or if this is what you're looking for), but turning it off with 
```python
self.log("loss_bootstrap_1", random_value, prog_bar=False)
```
once at a point your training phase changes might probably work.

**Reply by ChristophReich1996** [reply]
Hi @akihironitta, thanks for your help, unfortunately, if I set `prog_bar` to `False`, after the respective training phase, Lightning throws a `MisconfigurationException`...

```
pytorch_lightning.utilities.exceptions.MisconfigurationException: You called `self.log(loss_bs_2, ...)` twice in `training_step` with different arguments. This is not allowed
```

I'm using `pytorch-lightning==1.6.4`.

**Reply by akihironitta** [reply]
@ChristophReich1996 Ah right... I don't think it's feasible with the current progress bar. A simple workaround is to give up using a progress bar 😆 and use one of the loggers supported by PL and see metrics of your interest in the browser.
